components/script/dom/extendableevent.rs

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::codegen::Bindings::EventBinding::EventMethods;
+use dom::bindings::codegen::Bindings::ExtendableEventBinding;
+use dom::bindings::error::Fallible;
+use dom::bindings::global::GlobalRef;
+use dom::bindings::inheritance::Castable;
+use dom::bindings::js::Root;
+use dom::bindings::reflector::reflect_dom_object;
+use dom::bindings::str::DOMString;
+use dom::event::Event;
+use js::jsapi::{HandleValue, JSContext};
+use string_cache::Atom;
+
+// https://w3c.github.io/ServiceWorker/#extendable-event
+#[dom_struct]
+pub struct ExtendableEvent {
+    event: Event,
+    extensions_allowed: bool
+}
+
+impl ExtendableEvent {
+    pub fn new_inherited() -> ExtendableEvent {
+        ExtendableEvent {
+            event: Event::new_inherited(),
+            extensions_allowed: true
+        }
+    }
+    pub fn new_uninitialized(global: GlobalRef) -> Root<ExtendableEvent> {
```

Comment:
I don't see any need to have the split between initialized/uninitialized. That's only for older event types that can be created in an uninitialized states.